CMakeLists.txt

```diff
@@ -549,6 +549,18 @@ elseif (${ARCH} STREQUAL "x86_64")
 
 elseif (${ARCH} STREQUAL "riscv64")
 
+   if ("$ENV{BOARD}" STREQUAL "")
```

Comment:
Not a full review, but you shouldn't access the `$BOARD` env variable like that, simply because if it changes on the next build, it will be ignored by CMake. The proper way to that instead is use a "sticky" variable, so that the value of the ENV variable is taken when CMake is run, but if it changes after that, the build will fail. See `define_env_cache_str_var()`. You just need to add an entry for BOARD in this block, after ARCH:
```
# BEGIN [Environment-controlled "sticky" cache variables]

# Main variables
define_env_cache_str_var(TCROOT_PARENT ${CMAKE_SOURCE_DIR})
define_env_cache_str_var(TCROOT ${TCROOT_PARENT}/toolchain2)
define_env_cache_str_var(ARCH ${DEFAULT_TARGET_ARCH})
define_env_cache_bool_var(TEST_GCOV)
define_env_cache_bool_var(KERNEL_GCOV)
````